Draft of the Empower your AI migrations to Quarkus with Skills blog (#2660)
quarkusio/quarkusio.github.io: _posts/2026-05-26-ai-skills-migration.adoc

=== Step 4: Module selection and execution

As preamble to execute the set of the instructions, AI will report that it will now read the different "references" files and load the https://github.com/quarkusio/skills/blob/main/skills/migrate-spring-to-quarkus/SKILL.md#step-3-execute-modules[modules] to get the instructions:

.Skill load the modules
image::05-skill-load-modules.png[alt="skill load modules",align=left, width=80%]

As mentioned at the beginning of the blog, the module's instructions will be selected only if the condition declared part of the GATE CHECK is passing.

TIP: By looking to the "Gate result", you can immediately see which module has been selected and is currently processed

.Gate JDK check
image::06-gate-check.png[alt="Gate JDK check",align=left, width=70%]

The agent populates a list of "check boxes" corresponding to the list of the modules and will change the symbols during the migration process.

.Module Build System
image::08-module-build-executed.png[alt="Module Build system",align=left, width=85%]

> Here are the messages that we got by example during the execution of the module "Code" where Spring annotations have been discovered within the java source files.

Reviewer: ```suggestion
Here are some example messages produced during the execution of the "Code" module when Spring annotations were detected in Java source files.
```

> NOTE: The messages will depend on the model selected and tokens window size !

Reviewer: ```suggestion
NOTE: The messages will depend on the selected model and tokens window size !
```

.Module Code
image::07-module-code-executed.png[alt="Module code executed",align=left, width=95%]

> NOTE: The module "web" is optional if the application to be migrated don't include, thymeleaf files, jsp and/or JavaScript assets

Reviewer: ```suggestion
NOTE: The "web" module is optional if the application to be migrated doesn't include, thymeleaf files, jsp and/or JavaScript assets
```

> NOTE: The module "web" is optional if the application to be migrated don't include, thymeleaf files, jsp and/or JavaScript assets

Reviewer: The actual module in the skill is called `frontend`, I think it's better to use this name so the reader can easily find it in the source code.

---
layout: post

> title: 'Empower your AI migrations to Quarkus with Skills.'

Reviewer 2: ```suggestion
title: 'Empower your AI migrations to Quarkus with Skills'
```

> date: 2026-05-29

Reviewer 2: Bump to today's date  for visibility

Author: Fixed: c9d384143
